inc/cable.class.php

```diff
@@ -0,0 +1,621 @@
+                             'rand'    => $rand_socket_model_front]);
+      echo "</span>";
+      echo "</td></tr>";
+
+      //Line to display dropdown socket
+      echo "<tr class='tab_bg_1'>";
+      echo "<td>".Socket::getTypeName(1)."</td>";
+      echo "<td>";
+      echo "<span id='show_rear_sockets_field'>";
+      Socket::dropdown(['name'      => 'sockets_id_endpoint_a',
+                        'value'     => $this->fields["sockets_id_endpoint_a"],
+                        'condition' => ['socketmodels_id'   => $this->fields['socketmodels_id_endpoint_a'],
+                                        'itemtype'          => $this->fields['itemtype_endpoint_a'],
+                                        'items_id'          => $this->fields['items_id_endpoint_a']]
+                        ]);
+      echo "</span>";
+      echo "</td>";
+
+      echo "<td>".Socket::getTypeName(1)."</td>";
+      echo "<td>";
+      echo "<span id='show_front_sockets_field'>";
+      Socket::dropdown(['name'      => 'sockets_id_endpoint_b',
+                        'value'     => $this->fields["sockets_id_endpoint_b"],
+                        'condition' => ['socketmodels_id'   => $this->fields['socketmodels_id_endpoint_b'],
+                                        'itemtype'          => $this->fields['itemtype_endpoint_b'],
+                                        'items_id'          => $this->fields['items_id_endpoint_b']]
+                        ]);
+      echo "</span>";
+      echo "</td></tr>";
+
+      //Line to display asset breadcrum (datacenter / dcroom / rack / position)
+      echo "<tr class='tab_bg_1'>";
+      echo "<td>".__('Position')."</td>";
+      echo "<td>";
+
+      echo "<span id='show_rear_asset_breadcrumb'>";
+      if ($this->fields['items_id_endpoint_a']) {
+         if (method_exists($this->fields['itemtype_endpoint_a'], 'getDcBreadcrumbSpecificValueToDisplay')) {
+            echo $this->fields['itemtype_endpoint_a']::getDcBreadcrumbSpecificValueToDisplay($this->fields['items_id_endpoint_a']);
+         }
+      }
+      echo "</span>";
+
+      //Listener to update breacrumb / socket
+      echo Html::scriptBlock("
+         //listener to remove socket selector and breadcrumb
+         $(document).on('change', '#dropdown_itemtype_endpoint_a".$rand_itemtype_endpoint_a."', function(e) {
+            $('#show_rear_asset_breadcrumb').empty();
+            $('#show_rear_sockets_field').empty();
+         });
+
+         //listener to refresh socket selector and breadcrumb
+         $(document).on('change', '#dropdown_items_id_endpoint_a".$rand_items_id_endpoint_a."', function(e) {
+            items_id = $('#dropdown_items_id_endpoint_a".$rand_items_id_endpoint_a."').find(':selected').val();
+            itemtype = $('#dropdown_itemtype_endpoint_a".$rand_itemtype_endpoint_a."').find(':selected').val();
+            socketmodels_id = $('#dropdown_socketmodels_id_endpoint_a".$rand_socket_model_rear."').find(':selected').val();
```

Comment:
You should use the `var` keyword to prevent these variables to be set in the global scope.
```suggestion
            var items_id = $('#dropdown_items_id_endpoint_a".$rand_items_id_endpoint_a."').find(':selected').val();
            var itemtype = $('#dropdown_itemtype_endpoint_a".$rand_itemtype_endpoint_a."').find(':selected').val();
            var socketmodels_id = $('#dropdown_socketmodels_id_endpoint_a".$rand_socket_model_rear."').find(':selected').val();
```